presto-main/src/main/java/com/facebook/presto/execution/SharedBuffer.java

```diff
@@ -489,12 +424,19 @@ public BufferInfo getInfo()
             checkState(!Thread.holdsLock(SharedBuffer.this), "Thread must NOT hold a lock on the %s", SharedBuffer.class.getSimpleName());
 
             long sequenceId = this.sequenceId.get();
+            PageBuffer pageBuffer = partitionToPageBuffer.get(partition);
```

Comment:
I think if you reorder the creation code, you can add the PageBuffer to the constructor.